### PR TITLE
Rusk profile env dir

### PR DESCRIPTION
**Pull request by CPerezz**
Resolves: #307

### CODE REVIEW

File: rusk-profile/src/lib.rs [hunk 1]
```diff
@@ -57,7 +57,10 @@ fn file_stem(p: &Path) -> Option<&str> {
 }
 
 pub fn get_rusk_profile_dir() -> Result<PathBuf, io::Error> {
-    if let Some(mut dir) = home_dir() {
+    if let Some(env_var) = option_env!("RUSK_PROFILE_PATH") {
+        fs::create_dir_all(env_var.clone())?;
```
Comment (vlopes11) [inline rusk-profile/src/lib.rs:61]: If the env var is set, it is safe to assume the dir is already created by the user. Besides, there may be a conflict with existing dirs.

Suggestion: remove this line

File: rusk/Makefile [hunk 2]
```diff
@@ -2,10 +2,10 @@ help: ## Display this help screen
 	@grep -h -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-15s\033[0m %s\n", $$1, $$2}'
 
 keys: ## Build circuit keys
-	cargo build --release
+	RUSK_PROFILE_PATH=./../ cargo build --release
```
Comment (vlopes11) [inline rusk/Makefile:5]: I don't understand what is the meaning of this but I would leave this option to the user. If he wants to use custom directories for the rusk keys, then he can:

`$ RUSK_PROFILE_PATH=/tmp/rusk make keys`

Suggestion: rollback this line

File: rusk/Makefile [hunk 3]
```diff
@@ -2,10 +2,10 @@ help: ## Display this help screen
...
 
 test: ## Run Rusk tests
-	@cargo test \
+	RUSK_PROFILE_PATH=./../ cargo test \
```
Comment (vlopes11) [inline rusk/Makefile:8]: Same as above

File: .github/workflows/dusk_ci.yml [hunk 4]
```diff
@@ -32,7 +32,7 @@ jobs:
         env:
           cache-name: cache-rusk-profile-v2
         with:
-          path: ~/.rusk
+          path: ./
```
Comment (vlopes11) [inline .github/workflows/dusk_ci.yml:35]: The default should be the home dir, right?

File: rusk-profile/src/lib.rs [hunk 5]
```diff
@@ -57,7 +57,12 @@ fn file_stem(p: &Path) -> Option<&str> {
 }
 
 pub fn get_rusk_profile_dir() -> Result<PathBuf, io::Error> {
-    if let Some(mut dir) = home_dir() {
+    if let Some(env_var) = option_env!("RUSK_PROFILE_PATH") {
+        let mut path = PathBuf::from(env_var);
+        path.push(".rusk");
+        fs::create_dir_all(&path)?;
+        Ok(path)
+    } else if let Some(mut dir) = home_dir() {
```
Comment (ZER0) [inline rusk-profile/src/lib.rs:65]: Unfortunately I cannot select lines that are not part of this PR.
However, you should be able to remove some code duplication having something like:
```rust
pub fn get_rusk_profile_dir() -> Result<PathBuf, io::Error> {
    option_env!("RUSK_PROFILE_PATH")
        .map_or(home_dir(), |e| Some(PathBuf::from(e)))
        .and_then(|mut p| {
            p.push("./rusk");
            fs::create_dir_all(&p).map(|_| p).ok()
        })
        .ok_or_else(|| {
            warn!("rusk-profile dir not found and impossible to create");
            io::Error::new(
                io::ErrorKind::NotFound,
                "User Profile Dir not found",
            )
        })
}
```
Maybe you can even improve that, but it's just to give you the gist.

File: rusk/build.rs [hunk 6]
```diff
@@ -88,6 +86,11 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     tracing::subscriber::set_global_default(subscriber)
         .expect("setting default subscriber failed");
 
+    // If `RUSK_PROFILE_PATH` is not set, panic.
```
Comment (ZER0) [inline rusk/build.rs:89]: Unnecessary comment, the lines below are saying the same thing.

**Review by ZER0 — APPROVED** [review]
LGTM with the changes mentioned.

File: rusk-profile/src/lib.rs (hunk 5 above)
Comment (CPerezz) [reply rusk-profile/src/lib.rs:65]: Yupp. That looks much better!!!
